lib/stellar/client.rb

```diff
@@ -58,10 +58,25 @@ def initialize(options)
 
     Contract Stellar::Account => Any
     def account_info(account)
-      account_id  = account.address
+      account_id = account.address
       @horizon.account(account_id:account_id)._get
     end
 
+    Contract Stellar::Account => nil
+    def load_account_signers(account)
```

Comment:
If users had to use `account_info` they'd have to construct the `AccountSigner` instances themselves, which isn't the end of the world, but it requires them to know the structure of the horizon response.

Now that I'm thinking about this more I think we do something similar to how the [Python SDK](https://github.com/StellarCN/py-stellar-base/blob/master/stellar_sdk/server.py#L318) handles it. We could add a `load_account` method to the `Client` class that returns an `Account` class that has its `signers` and `thresholds` populated. We could make `Account.signers` and `Account.thresholds` immutable and just pass the values to the constructor.

Reply:
This has been implemented

Reply:
I still think this idea of mutating the Account object is not intuitive because someone can have an `Account` object and its thresholds and signers aren't set, and it isn't obvious that you have to use a different method to get those values set. I think it's really intuitive that the account_info function already returns this data we need, we're just wanting it to be massaged into a better/easier to use format?

Here's a way we could do it where the user calls `account_info` that already exists to get the additional info, and we provide additional functions on that type to make using that information easier to address the ease of use issue.

```ruby
class AccountSigner < Hash
  def initialize(params)
    self.merge!(params)
  end

  def key
    self['key']
  end

  def weight
    self['weight']
  end
end

class AccountInfo < Hash
  def initialize(params)
    self.merge!(params)
  end

  def thresholds
    self['thresholds']
  end

  def signers
    self['signers'].map do |signer|
      AccountSigner.new(signer)
    end
  end
end

def account_info
  info = { # fake info that we would have gotten from Horizon
    'thresholds' => [1,2,3,4],
    'signers' => [
      { 'key' => 'G1...', 'weight' => 1 },
      { 'key' => 'G2...', 'weight' => 1 },
    ],
  }
  AccountInfo.new(info)
end

ai = account_info
puts ai
puts ai.thresholds.inspect
puts ai.signers[0].key
```

This is backwards compatible because the types returned as still hashes and still contain all the other details, we're just upgrading them and including some additional information.

Reply:
Ok, I think this works. I do feel like there is a rather strong precedent in this SDK to add a `address` method to `AccountSigner` as well, considering `KeyPair.address` and `Account.address` already exist.

Reply:
Looks like the base repository has a `Hash` type declared within the `Stellar` module thats conflicting with the `AccountSigner` and `AccountInfo` declaration. I can get around it by doing something like this:
```
$Hash = Hash
module Stellar
    class AccountInfo < $Hash
        ...
    end
end
```
Is there a better way to do this without using a global variable?
```

Reply:
Ah, yes we can do that another way. Reference the built-in `Hash` type by adding a `::` to the front of it so you're telling it you want the Hash in the global namespace.

```ruby
class AccountInfo < ::Hash
```

Reply:
After trying to implement this I've realized its not a backwards compatible solution, because `account_info` returns a `HyperClient::Resource` object, not a `Hash`. Users of the SDK access the data using attributes on the object, so like:
```
client = Stellar::Client.default_testnet
info = client.account_info(a)  # assuming 'a' is an account object
p info.sequence
```

Reply:
That's awesome then. It means we don't need to create any of the objects. You're going to get back an object that you can call `.signers[].key` on and just pass that around.

Reply:
Do we still want to use Sets instead of Array's?